content/intro-to-storybook/react/en/data.md

```diff
@@ -166,40 +194,107 @@ Default.args = {
     { ...TaskStories.Default.args.task, id: '5', title: 'Task 5' },
     { ...TaskStories.Default.args.task, id: '6', title: 'Task 6' },
   ],
+  status: 'idle',
+  error: null,
 };
 
-export const WithPinnedTasks = Template.bind({});
-WithPinnedTasks.args = {
-  // Shaping the stories through args composition.
-  // Inherited data coming from the Default story.
-  tasks: [
-    ...Default.args.tasks.slice(0, 5),
-    { id: '6', title: 'Task 6 (pinned)', state: 'TASK_PINNED' },
-  ],
+// A super-simple mock of a redux store
+const Mockstore = ({ taskboxState, children }) => (
+  <Provider
+    store={configureStore({
+      reducer: {
+        taskbox: createSlice({
+          name: 'taskbox',
+          initialState: taskboxState,
+          reducers: {
+            updateTaskState: (state, action) => {
+              const { id, newTaskState } = action.payload;
+              const task = state.tasks.findIndex((task) => task.id === id);
+              if (task >= 0) {
+                state.tasks[task].state = newTaskState;
+              }
+            },
+          },
+        }).reducer,
+      },
+    })}
+  >
+    {children}
+  </Provider>
+);
+
+export default {
+  component: TaskList,
+  title: 'TaskList',
+  decorators: [(story) => <div style={{ padding: "3rem" }}>{story()}</div>],
 };
 
+const Template = () => <TaskList />;
+
+export const Default = Template.bind({});
+Default.decorators = [
```

Comment:
If you wanted to reduce some of the boilerplate here[^1], you could write a meta-level decorator like:

```jsx
(story, { parameters }) => <Mockstore taskboxState={parameters.taskboxState}>{story()}</Mockstore>
```

Then each story just needs to pass the state:

```js
export const Default = Template.bind({});
Default.parameters = { taskboxState: MockedState };

const pinnedtasks = [
  ...MockedState.tasks.slice(0, 5),
  { id: '6', title: 'Task 6 (pinned)', state: 'TASK_PINNED' },
];
export const WithPinnedTasks = Template.bind({});
WithPinnedTasks.parameters = {
  taskboxState: {
    ...MockedState,
    tasks: pinnedtasks,
  }
};
```

[^1]: Sometimes it's less "noisy" to avoid repetition and focus only on what changes between instances. But sometimes the extra abstraction can cause confusion, especially for early learners. 🤷‍♂️